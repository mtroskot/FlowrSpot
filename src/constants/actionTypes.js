export const userActionTypes = Object.freeze({
  REGISTER_USER: 'REGISTER_USER',
  LOGIN_USER: 'LOGIN_USER',
  LOGIN_USER_SUCCESS: 'LOGIN_USER_SUCCESS',
  LOGOUT_USER: 'LOGOUT_USER'
});

export const flowerActionTypes = Object.freeze({
  GET_FAVORITE_FLOWERS: 'GET_FAVORITE_FLOWERS',
  GET_FAVORITE_FLOWERS_SUCCESS: 'GET_FAVORITE_FLOWERS_SUCCESS',
  FAVORITE_FLOWER: 'FAVORITE_FLOWER',
  FAVORITE_FLOWER_SUCCESS: 'FAVORITE_FLOWER_SUCCESS',
  UNFAVORITE_FLOWER: 'UNFAVORITE_FLOWER',
  UNFAVORITE_FLOWER_SUCCESS: 'UNFAVORITE_FLOWER_SUCCESS'
});

export const uiActionTypes = Object.freeze({
  START_ACTION: 'START_ACTION',
  STOP_ACTION: 'STOP_ACTION',
  REFRESH_ACTION_START: 'REFRESH_ACTION_START',
  REFRESH_ACTION_STOP: 'REFRESH_ACTION_STOP',
  TOGGLE_POPUP_MESSAGE: 'TOGGLE_POPUP_MESSAGE'
});
