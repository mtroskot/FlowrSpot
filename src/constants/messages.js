export const DEFAULT_ERROR = 'Something went wrong';
export const NO_RESULTS = 'No Results Found';
export const REGISTRATION_SUCCESS = 'Registration Success';
