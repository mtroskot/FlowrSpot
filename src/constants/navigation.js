export const screenNames = Object.freeze({
  COMMENTS: 'COMMENTS',
  FLOWER_DETAILS: 'FLOWER_DETAILS',
  FLOWERS: 'FLOWERS',
  LATEST_SIGHTINGS: 'LATEST_SIGHTINGS',
  FAVORITES: 'FAVORITES',
  SETTINGS: 'SETTINGS',
  LOGIN: 'LOGIN',
  REGISTER: 'REGISTER',
  HOME: 'HOME',
  LOCATION: 'LOCATION',
  MAP: 'MAP',
  MENU: 'MENU'
});

export const stackNames = Object.freeze({
  COMMENTS_STACK: 'COMMENTS_STACK',
  HOME_STACK: 'HOME_STACK',
  LOCATION_STACK: 'LOCATION_STACK',
  MAP_STACK: 'MAP_STACK',
  MENU_STACK: 'MENU_STACK'
});
