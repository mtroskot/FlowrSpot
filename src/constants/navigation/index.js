export const screenNames = Object.freeze({
  HOME: 'HOME'
});
