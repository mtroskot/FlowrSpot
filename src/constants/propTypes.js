import PropTypes from 'prop-types';

export const flowerPropTypes = PropTypes.exact({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  latin_name: PropTypes.string.isRequired,
  sightings: PropTypes.number.isRequired,
  profile_picture: PropTypes.string.isRequired,
  favorite: PropTypes.bool.isRequired
});

export const userPropTypes = PropTypes.exact({
  id: PropTypes.number,
  email: PropTypes.string,
  authToken: PropTypes.string,
  firstName: PropTypes.number,
  lastName: PropTypes.string
});

export const popupMessagePropTypes = PropTypes.exact({
  message: PropTypes.string,
  position: PropTypes.oneOf(['bottom', 'top'])
});

export const favoriteFlowerListPropTypes = PropTypes.arrayOf(
  PropTypes.exact({
    flowerId: PropTypes.number.isRequired,
    favoriteId: PropTypes.number.isRequired
  })
);
