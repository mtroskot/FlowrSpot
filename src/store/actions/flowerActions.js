import { flowerActionTypes } from 'src/constants/actionTypes';

export const getFavoriteFlowers = () => ({
  type: flowerActionTypes.GET_FAVORITE_FLOWERS,
  payload: {}
});

export const getFavoriteFlowersSuccess = favoriteFlowers => ({
  type: flowerActionTypes.GET_FAVORITE_FLOWERS_SUCCESS,
  payload: { favoriteFlowers }
});

export const favoriteFlower = flowerId => ({
  type: flowerActionTypes.FAVORITE_FLOWER,
  payload: { flowerId }
});

export const favoriteFlowerSuccess = (flowerId, favoriteId) => ({
  type: flowerActionTypes.FAVORITE_FLOWER_SUCCESS,
  payload: { flowerId, favoriteId }
});

export const unfavoriteFlower = (flowerId, favoriteId) => ({
  type: flowerActionTypes.UNFAVORITE_FLOWER,
  payload: { flowerId, favoriteId }
});

export const unfavoriteFlowerSuccess = flowerId => ({
  type: flowerActionTypes.UNFAVORITE_FLOWER_SUCCESS,
  payload: { flowerId }
});
