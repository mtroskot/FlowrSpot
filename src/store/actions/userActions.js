import { userActionTypes } from 'src/constants/actionTypes';

export const registerUser = registerData => ({
  type: userActionTypes.REGISTER_USER,
  payload: { registerData }
});

export const loginUser = (email, password) => ({
  type: userActionTypes.LOGIN_USER,
  payload: { email, password }
});

export const loginUserSuccess = (id, email, authToken, firstName, lastName) => ({
  type: userActionTypes.LOGIN_USER_SUCCESS,
  payload: { id, email, authToken, firstName, lastName }
});

export const logoutUser = () => ({
  type: userActionTypes.LOGOUT_USER,
  payload: {}
});
