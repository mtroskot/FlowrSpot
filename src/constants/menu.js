import { screenNames } from 'src/constants/navigation';

export const menuList = Object.freeze([
  {
    menuId: screenNames.FLOWERS,
    text: 'Flowers'
  },
  {
    menuId: screenNames.LATEST_SIGHTINGS,
    text: 'Latest Sightings'
  },
  {
    menuId: screenNames.FAVORITES,
    text: 'Favorites'
  },
  {
    menuId: screenNames.SETTINGS,
    text: 'Settings'
  },
  {
    menuId: screenNames.LOGIN,
    text: 'Login'
  }
]);
