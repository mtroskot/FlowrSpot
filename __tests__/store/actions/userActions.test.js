import configureStore from 'redux-mock-store';
import * as userActions from 'src/store/actions/userActions';
import { userActionTypes } from 'src/constants/actionTypes';

const middlewares = [];
const mockStore = configureStore(middlewares);
// Initialize mockstore with empty state
const initialState = {};
const store = mockStore(initialState);

describe('User actions tests', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('dispatches registerUser action', () => {
    const registerData = {};
    const expectedActions = [
      {
        type: userActionTypes.REGISTER_USER,
        payload: { registerData }
      }
    ];
    store.dispatch(userActions.registerUser(registerData));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches loginUser action', () => {
    const email = 'email';
    const password = 'password';
    const expectedActions = [
      {
        type: userActionTypes.LOGIN_USER,
        payload: { email, password }
      }
    ];
    store.dispatch(userActions.loginUser(email, password));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
  it('dispatches loginUserSuccess action', () => {
    const id = 'id';
    const email = 'email';
    const authToken = 'authToken';
    const firstName = 'firstName';
    const lastName = 'lastName';
    const expectedActions = [
      {
        type: userActionTypes.LOGIN_USER_SUCCESS,
        payload: { id, email, authToken, firstName, lastName }
      }
    ];
    store.dispatch(userActions.loginUserSuccess(id, email, authToken, firstName, lastName));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
  it('dispatches logoutUser action', () => {
    const expectedActions = [
      {
        type: userActionTypes.LOGOUT_USER,
        payload: {}
      }
    ];
    store.dispatch(userActions.logoutUser());
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
});
