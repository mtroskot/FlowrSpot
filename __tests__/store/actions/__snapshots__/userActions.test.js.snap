// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User actions tests dispatches loginUser action 1`] = `
Array [
  Object {
    "payload": Object {
      "email": "email",
      "password": "password",
    },
    "type": "LOGIN_USER",
  },
]
`;

exports[`User actions tests dispatches loginUserSuccess action 1`] = `
Array [
  Object {
    "payload": Object {
      "authToken": "authToken",
      "email": "email",
      "firstName": "firstName",
      "id": "id",
      "lastName": "lastName",
    },
    "type": "LOGIN_USER_SUCCESS",
  },
]
`;

exports[`User actions tests dispatches logoutUser action 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "LOGOUT_USER",
  },
]
`;

exports[`User actions tests dispatches registerUser action 1`] = `
Array [
  Object {
    "payload": Object {
      "registerData": Object {},
    },
    "type": "REGISTER_USER",
  },
]
`;
